beater/server.go

```diff
@@ -62,12 +62,18 @@ func newServer(config Config, publish successCallback) *http.Server {
 	}
 }
 
-func start(server *http.Server, ssl *SSLConfig) {
-	if ssl.isEnabled() {
-		go server.ListenAndServeTLS(ssl.Cert, ssl.PrivateKey)
-	} else {
-		go server.ListenAndServe()
-	}
+func start(server *http.Server, ssl *SSLConfig) chan(error) {
+	c := make(chan(error), 1)
```

Comment:
Instead of having the go routines in here, we could call `start` inside a go routine. Like this we would not need an error channel but could directly handle it in `beater.go`. Something like:

```
go func() {
	err := start(..., ...)
	if err != nil {
		logp.Err(err)
		close(bt.done)
	}

}

<-bt.done
```

In this case I would probably rename the function from `start` to `run`.